starlette/requests.py

```diff
@@ -121,11 +130,18 @@ def url_for(self, name: str, **path_params: typing.Any) -> str:
     raise RuntimeError("Receive channel has not been made available")
 
 
+async def empty_send(message: Message) -> None:
+    raise RuntimeError("Send channel has not been made available")
+
+
 class Request(HTTPConnection):
-    def __init__(self, scope: Scope, receive: Receive = empty_receive):
+    def __init__(
+        self, scope: Scope, receive: Receive = empty_receive, send: Send = empty_send
+    ):
```

Comment:
I guess we *might* want to think about what we'd need to do to move this up to the top `HTTPConnection` instead, since both `Request` and `WebSocket` now accept `scope`, `receive`, `send`.

We *could* consider having public `def send()` and `def receive()` methods on the HTTPConnection (eg. the WebSocket implementations that add in state transition checks would then call into `super()` for the raw sending.)

Any thoughts on that?

Reply:
My policy is usually "keep it private until you absolutely need to make it public"  :)

Reply:
Yes - that's entirely fair here.